### PR TITLE
Custom effect example

### PR DESCRIPTION
**Pull request by squarewavesbelow**
This is a very simple utility  effect, with a gain knob and a mute switch for input channels. Purpose of this PR is to reference it in a basic wiki tutorial/guide on  effects development for mixxx.

### CODE REVIEW

**Comment by Be-ing** [comment]
Do you want to have this merged or leave it as an unmerged pull request that can be referenced from a tutorial? I think we could merge it if the lines registering this effect in src/effects/native/nativebackend.cpp and builds/depends.py are commented out.

**Comment by squarewavesbelow** [comment]
I just wrote it to be referenced in the wiki article of a mixxx effect development guide. To be honest I found it quite useful as a quick and dirty utility effect to mute the effect chain I am working on, and if there is interest in such a utility effect (or a set of utility effects i.e mute/dc/gain/panning/phase) it would be meaningful to be further developed and merged.
 Either way I do also believe that example effect is a more suitable name, so I will change naming and commit back.

**Comment by Be-ing** [comment]
I think it will be helpful to merge it. If it is left out of the upstream code base it will likely not get updated as changes are made to the effects system.

**Comment by daschuer** [comment]
I don't think we need it in this stage in the code base. We have already other simple effects like bit rushed, that can be treated as an example. If you think we have a use case for a gain effect, we can include it. But this requires a popper naming here and ramping to avoid crackling. 
I can also think of other one liner effects that could be usefull for learning effects, for example a tanh limiter, or other transpose effects.

File: src/test/nativeeffects_test.cpp [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #include "effects/native/bessel4lvmixeqeffect.h"
 #include "effects/native/bessel8lvmixeqeffect.h"
 #include "effects/native/bitcrushereffect.h"
+#include "../effects/native/utilityeffect.h"
```
Comment (daschuer) [inline src/test/nativeeffects_test.cpp]: please use includes relative to the src directory, without ../

File: src/effects/native/utilityeffect.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+QString UtilityEffect::getId() {
+    return "org.mixxx.effects.utility";
+}
+
+// static
+EffectManifest UtilityEffect::getManifest() {
+    EffectManifest manifest;
+    manifest.setId(getId());
+    manifest.setName(QObject::tr("Utility"));
+    manifest.setAuthor("sqrwvzblw");
```
Comment (daschuer) [inline src/effects/native/utilityeffect.cpp]: hää? I think we use "The Mixxx Team" in the other effects.

File: src/effects/native/utilityeffect.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+	//increase and decrease ramp factor from 0.0 to 1.0 and back with this step.
+	CSAMPLE ramp_step = (pState->ramp_time_in_seconds * numSamples)/(2.0*sampleRate);
+
+	UtilityEffect::rampGainCalculation(pState, current_mute_state, ramp_step, enableState );
+
+
+    const int kChannels = 2;
+    for (unsigned int i = 0; i < numSamples; i += kChannels) {
```
Comment (daschuer) [inline src/effects/native/utilityeffect.cpp]: we have vectorized functions for this case. SampleUtil::copyWithRampingGain() or similar. Pleas try to make use of this.

**Comment by daschuer** [comment]
Why is it now called Utility and not just Gain?

**Comment by Be-ing** [comment]
This has not been updated in a long time and it is not clear that this is something that should be merged into the Mixxx code, so I am closing this PR.

**Pull request closed without merging**